core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/BuildSystemTargetBuildItem.java

```diff
@@ -11,10 +11,12 @@
 
     private final Path outputDirectory;
     private final String baseName;
+    private final boolean rebuild;
 
-    public BuildSystemTargetBuildItem(Path outputDirectory, String baseName) {
+    public BuildSystemTargetBuildItem(Path outputDirectory, String baseName, boolean rebuild) {
```

Comment:
What are the reason for the rebuild Boolean a. Is it just to read from the dat model? Or is there more to it ?

Reply:
When you rebuild you don't want to re-copy libraries, as they are already in place on disk. A normal build will also delete the target dir to start with a clean slate.